### ISSUE TITLE
Requirements for trusted certificates in macOS 10.15

### ISSUE THREAD

**Issue by benarent** [gravitational/teleport#4163]
### Description

**What happened**:

While working on the quickstart, I'm still running into issues locally. Having Subject Alternative Name has helped for setup but I'm still running into an issue with Chrome really not trusting it.

We've most of the requirements https://support.apple.com/en-us/HT210176 but need 

- ExtendedKeyUsage
- Reducing self signed certs to 800 days 

> TLS server certificates must contain an ExtendedKeyUsage (EKU) extension containing the id-kp-serverAuth OID 

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            da:37:ae:b7:46:27:d7:64:54:24:3d:6f:99:ad:48:d8
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=US, O=localhost, CN=localhost
        Validity
            Not Before: Aug  3 21:26:25 2020 GMT
            Not After : Aug  1 21:26:25 2030 GMT
        Subject: C=US, O=localhost, CN=localhost
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:cb:bf:aa:d7:f9:b2:6b:da:d1:89:67:6c:53:56:
                    53:b9:6c:3b:dd:83:60:8b:ed:9c:e8:b5:29:c4:36:
                    4a:2a:ae:dd:c0:0b:0b:af:88:6d:50:3f:0c:61:c3:
                    bd:53:13:c8:68:18:67:7e:2a:f6:44:b1:53:64:e8:
                    48:3f:e1:f4:85:45:65:98:a6:26:bc:ce:69:79:58:
                    0d:0e:d1:29:2d:a5:79:bb:84:1e:4f:04:f5:cb:28:
                    c8:18:a7:45:7c:ad:b3:4e:3e:54:e3:ab:ab:63:aa:
                    a9:03:f2:5b:2d:b8:02:a9:fa:ce:62:fc:2e:ff:f6:
                    23:9e:62:73:86:4b:97:d2:7b:f5:e8:8f:01:77:ff:
                    05:86:03:3f:82:23:99:0c:09:af:24:00:68:1f:5d:
                    0f:bf:6c:44:e4:97:a5:d4:84:53:cd:2e:0c:c5:bf:
                    f0:08:ac:87:b7:5b:7d:a8:e4:3b:b0:7a:12:c6:01:
                    3f:d5:8f:67:db:cd:40:ca:d7:d3:37:ad:e9:ad:7d:
                    f2:2e:e3:de:59:83:bf:e3:36:9e:b3:17:37:9f:3d:
                    c2:27:64:38:34:b5:88:9b:cf:dc:d1:c3:ff:da:a7:
                    2a:3b:12:d6:58:e2:f0:6a:75:bb:eb:c8:4a:19:0b:
                    97:a5:b2:7f:32:97:b2:bd:a9:bd:0e:d1:7e:4b:52:
                    ae:a9
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment, Certificate Sign
            X509v3 Basic Constraints: critical
                CA:TRUE
            X509v3 Subject Alternative Name:
                DNS:Bens-MacBook-Pro.local, DNS:localhost, DNS:localhost.local, IP Address:127.0.0.1, IP Address:0:0:0:0:0:0:0:1
    Signature Algorithm: sha256WithRSAEncryption
         b2:a9:e5:e9:22:16:5a:0b:2f:8a:ab:2d:75:7c:04:a5:c7:19:
         55:bc:a8:e4:c3:5a:fb:50:d0:1a:f3:96:bd:82:64:b5:75:4b:
         6c:5f:d0:6e:83:54:46:48:69:5f:fd:60:69:a9:bf:a6:2f:f8:
         68:e8:49:6e:00:2b:39:e3:7c:c0:ea:ae:f6:46:b3:5b:66:0f:
         40:fc:67:37:ad:3b:72:6c:50:ff:a2:60:ee:5f:e6:4d:5e:90:
         63:ca:5b:f1:d6:56:4c:41:a3:88:ca:f2:44:1f:34:c0:61:ca:
         1f:d3:e0:33:8b:e5:88:64:6c:99:4e:c6:53:6e:09:78:d0:fe:
         f4:8d:23:e5:de:84:50:d8:5d:89:af:50:3d:43:91:f7:1f:d1:
         8c:3d:50:c2:83:5a:81:30:cc:4b:c6:90:12:54:32:59:17:bf:
         84:ea:80:0c:1a:92:7f:d3:ca:9e:29:06:28:2a:72:6c:93:e6:
         e2:e2:e4:5e:4e:03:cf:59:e5:47:8a:18:83:9d:d6:9d:84:84:
         a6:cb:bc:eb:0a:17:a0:58:c9:48:c0:67:11:9b:47:f3:8f:b3:
         be:61:65:ef:80:09:9c:4f:8e:07:4f:e2:43:0a:48:30:b5:71:
         ee:f4:a4:f4:a9:95:28:31:5f:fc:77:47:6f:24:ce:fe:fc:2a:
         0a:83:5f:9c
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

### Environment
- Teleport version (use `teleport version`):
- Tsh version (use `tsh version`):
- OS (e.g. from `/etc/os-release`):

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware):

**Browser environment**

- Browser Version (for UI-related issues):
- Install tools:
- Others:

**Relevant Debug Logs If Applicable**

- `tsh --debug`
- `teleport --debug`

**Comment by webvictim** [comment]
This is specifically a Safari thing, right? Can we ask people to use Chrome in the meantime?

**Comment by lenko-d** [comment]
Similar issue was fixed in Gravity and the fix could be re-used:

[Gravity self signed certificates don't meet requirements MacOS Catalina](https://github.com/gravitational/gravity/issues/868)

**Comment by zmb3** [comment]
Fixed with f6bb323
